include/glow/Graph/Graph.h

```diff
@@ -627,6 +627,56 @@ class Function final : public Named {
   ARITHMETIC_FUN_DECL(Pow);
 #undef ARITHMETIC_FUN_DECL
 
+  std::vector<NodeValue>
+  broadcastInputs(int axis, const llvm::ArrayRef<NodeValue> inputs);
+
+  template <class T, class U>
+  using enable_if_same_t = std::enable_if<std::is_same<T, U>::value, U>;
+
+#define BROADCAT_FUNC_COMMON_CODE(NUM_INPUTS)                                  \
+  constexpr size_t numInputs = sizeof...(Args);                                \
+  static_assert(numInputs == NUM_INPUTS,                                       \
+                "Invalid input passed in to commonCreateBroadcast.");          \
+  std::vector<NodeValue> inputs = broadcastInputs(axis, {inputArgs...});
+
+  /// Template function that creates a node and normalizes it's input shapes
+  /// with the use of BroadCast nodes. If axis is -1, it calculates it
+  /// automatically.
+  template <class T, class... Args>
```

Comment:
nit: These are all boilerplate for their respective Nodes -- IMHO it's cleaner to just add a single macro for them and then do e.g. `DECLARE_BROADCAST_ARITHMETIC(Mul)` etc. This is also consistent with what we do with other cases for Arithmetic nodes.

Reply:
@jfix71 I had something like this originally. Similar to what I did in GraphOptimizer.cpp @bertmaher mentioned it was not clear. Although I lean more towards macro I don't have a strong opinion one way or the other on this. Both approaches have their plus and minuses.

Reply:
@jfix71 if you prefer the macro that's fine.  I guess opinions are different here :-)

Reply:
Huh, I thought you wanted the macro with just the extra template magic removed. I meant a macro for the entire Arithmetic Node, e.g. something like:

```cpp
#define DECLARE_BROADCAST_ARITHMETIC(NODE_NAME)              \
  template <class T, class... Args>                          \
  typename enable_if_same_t<T, NODE_NAME##Node>::type *      \
  createNodeWithBroadcast(const std::string &name, int axis, \
                          Args &&... inputArgs) {            \
    BROADCAST_FUNC_COMMON_CODE(/* NUM_INPUTS */ 2)           \
    return create##NODE_NAME(name, inputs[0].getType(),      \
                             inputs[0], inputs[1]);          \
  }
DECLARE_BROADCAST_ARITHMETIC(Mul)
DECLARE_BROADCAST_ARITHMETIC(Div)
DECLARE_BROADCAST_ARITHMETIC(Add)
DECLARE_BROADCAST_ARITHMETIC(Sub)
```
I'm fine either way though 🙂

Reply:
Ah. I see. Yeah this looks good. I'll change to this. :)

Reply:
Modified slightly to make it more generic to support Where/Select down the road. Probably Less/cmpLT.